File: 2015/01 Not quite LISP/solution.py
def floor(s):
    return s.count("(") - s.count(")")

def first_basement(s):
    f = 0
    for i, e in enumerate(s):
        if e == "(":
            f += 1
        else:
            f -= 1
        if f < 0:
            return i + 1

seq = open("input.txt").read().strip()
print(floor(seq))
print(first_basement(seq))

File: 2015/01 Not quite LISP/solution.rb
data = IO.read("input.txt").strip
floor = 0
first_b = -1
data.each_char.with_index do |c, i|
    floor += c == "(" ? 1 : -1
    if floor < 0 and first_b < 0
        first_b = i + 1
    end
    print i, floor, "\n"
end
print floor, "\n"
print first_b, "\n"
